pkg/providers/instancetype/instancetype.go

```diff
@@ -177,7 +177,7 @@ func (p *Provider) getInstanceTypeZones(ctx context.Context, nodeTemplate *v1alp
 		return nil, err
 	}
 	if len(subnets) == 0 {
-		return nil, fmt.Errorf("no subnets matched selector %v", nodeTemplate.Spec.SubnetSelector)
+		return nil, nil
```

Comment:
Assuming we log as part of the nodetemplatecontroller?

Reply:
yep, I logged here first and then realized how many times this things is called